boards/arm/atsamr21_xpro/atsamr21_xpro_defconfig

```diff
@@ -0,0 +1,23 @@
+# Copyright (c) 2018 Bryan O'Donoghue
+# SPDX-License-Identifier: Apache-2.0
+CONFIG_ARM=y
+CONFIG_SOC_FAMILY_SAM0=y
+CONFIG_SOC_SERIES_SAMR21=y
+CONFIG_SOC_PART_NUMBER_SAMR21G18A=y
+CONFIG_BOARD_ATSAMR21_XPRO=y
+CONFIG_CORTEX_M_SYSTICK=y
+CONFIG_CONSOLE=y
+CONFIG_UART_CONSOLE=y
+CONFIG_SERIAL=y
+CONFIG_UART_SAM0=y
+CONFIG_UART_INTERRUPT_DRIVEN=y
+CONFIG_GPIO=y
+CONFIG_GPIO_SAM0=y
+CONFIG_WATCHDOG=y
+CONFIG_WDT_SAM0=y
+CONFIG_SOC_ATMEL_SAMD_XOSC32K=y
+CONFIG_SOC_ATMEL_SAMD_XOSC32K_AS_MAIN=y
+# CONFIG_SPI is not set
+# CONFIG_SPI_SAM0 is not set
```

Comment:
Unused code should be removed. However, board documentation claims that SPI is supported.

Reply:
I could create a samr21.dtsi with
```
#include <atmel/samd21.dtsi>
&sercom4 {
       status = "ok";
       compatible = "atmel,sam0-spi";
       dipo = <0>;
       dopo = <2>;
};
```
which is the internal SPI where the radio is attached to like in [nxp_kw2xd.dtsi](https://github.com/zephyrproject-rtos/zephyr/blob/master/dts/arm/nxp/nxp_kw2xd.dtsi#L249). There a at86rf233@0 node could be attached later.

I could also just map one of the remaining SerComs to SPI on the debug header of the board.

Reply:
If the code is not used it should be removed. Also - the rules are a bit fuzzy here - I believe that only basic functionality should be enabled in this file. @galak will know the definite answer.

Reply:
Ok, I removed it for now until, can add it back later when support for the radio is in place.